pkg/controller/directvolumemigration/stunnel.go

```diff
@@ -0,0 +1,495 @@
+		OrganizationalUnit: []string{"Engineering"},
+	}
+
+	certTemp := x509.Certificate{
+		SerialNumber:          big.NewInt(2020),
+		Subject:               subj,
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(10, 0, 0),
+		IsCA:                  true,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+
+	caBytes, err := x509.CreateCertificate(
+		rand.Reader,
+		&certTemp,
+		&certTemp,
+		&caPrivKey.PublicKey,
+		caPrivKey,
+	)
+	if err != nil {
+		return err
+	}
+
+	caPEM := new(bytes.Buffer)
+	err = pem.Encode(caPEM, &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: caBytes,
+	})
+	if err != nil {
+		return err
+	}
+
+	caPrivKeyPEM := new(bytes.Buffer)
+	err = pem.Encode(caPrivKeyPEM, &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(caPrivKey),
+	})
+	if err != nil {
+		return err
+	}
+
+	// Create secret in each namespace  src+dest with tls.crt = caPEM and tls.key
+	// = caPrivKeyPEM
+	// Secret data contains:
+	// ca.crt
+	// tls.crt (right now equal to ca.crt)
+	// tls.key
+
+	pvcMap := t.getPVCNamespaceMap()
+	for ns, _ := range pvcMap {
+		srcSecret := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: ns,
+				Name:      "directvolumemigration-stunnel-certs",
+				Labels: map[string]string{
+					"app": "directvolumemigration-rsync-transfer",
+				},
+			},
+			Data: map[string][]byte{
+				"tls.crt": caPEM.Bytes(),
+				"ca.crt":  caPEM.Bytes(),
+				"tls.key": caPrivKeyPEM.Bytes(),
+			},
+		}
+		destSecret := srcSecret
+		err = srcClient.Create(context.TODO(), &srcSecret)
+		if k8serror.IsAlreadyExists(err) {
+			t.Log.Info("Secret already exists on source", "namespace", srcSecret.Namespace)
+		} else if err != nil {
+			return err
+		}
+		err = destClient.Create(context.TODO(), &destSecret)
+		if k8serror.IsAlreadyExists(err) {
+			t.Log.Info("Secret already exists on destination", "namespace", destSecret.Namespace)
+		} else if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Create stunnel client pods + svc
+func (t *Task) createStunnelClientPods() error {
+	srcClient, err := t.getSourceClient()
+	if err != nil {
+		return err
+	}
+	pvcMap := t.getPVCNamespaceMap()
+	for ns, _ := range pvcMap {
+		svc := corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "directvolumemigration-rsync-transfer-svc",
+				Namespace: ns,
+				Labels: map[string]string{
+					"app": "directvolumemigration-rsync-transfer",
+				},
+			},
+			Spec: corev1.ServiceSpec{
+				Ports: []corev1.ServicePort{
+					{
+						Name:       "stunnel",
+						Protocol:   corev1.ProtocolTCP,
+						Port:       int32(2222),
+						TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: 2222},
+					},
+				},
+				Selector: map[string]string{
+					"app":     "directvolumemigration-rsync-transfer",
```

Comment:
probably worth creatint a global variable for labels on service/pod/route for easier maintenance

Reply:
let's also use correlation labels to make this more consistent with existing workflows.